### DISCUSSION TITLE
Login username or email

### DISCUSSION THREAD

**Discussion by dinhhieu67** [codeigniter4/shield#1100] [Q&A]
Hi all!
I want the system to log in either by username or email. Please help!

**Comment by kenjis** [comment]
https://github.com/codeigniter4/shield/blob/3fbac7a3da41ebaac7d51708244f68df596409f0/src/Config/Auth.php#L275-L284

**Reply by warcooft** [reply]
Hi @kenjis is it possible if we use both at the same time? 
using email or username in one column, simply, Shield will check based on email or username at the same time.

**Reply by dinhhieu67** [reply]
@kenjis  I want to log in like Facebook, email or phone number can log in. 
<img width="390" alt="Screenshot 2024-04-16 at 10 16 24" src="https://github.com/codeigniter4/shield/assets/8575385/1b36c749-5d6a-488a-b334-63df0913acd2">

**Reply by kenjis** [reply]
Shield does not support one field with email or username (email or phone number).
If you want it, customize Shield.
